pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -17,19 +17,30 @@
 from typing import List, Optional, Union
 from torch.utils.data import DataLoader
 from pytorch_lightning.utilities.model_utils import is_overridden
+from pytorch_lightning.utilities import rank_zero_warn
 
 
 class DataConnector(object):
 
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, check_val_every_n_epoch, reload_dataloaders_every_epoch, prepare_data_per_node):
+    def on_trainer_init(self, check_val_every_n_epoch, reload_dataloaders_every_n_epochs, reload_dataloaders_every_epoch, prepare_data_per_node):
         self.trainer.datamodule = None
         self.trainer.prepare_data_per_node = prepare_data_per_node
 
         self.trainer.check_val_every_n_epoch = check_val_every_n_epoch
-        self.trainer.reload_dataloaders_every_epoch = reload_dataloaders_every_epoch
+
+        if reload_dataloaders_every_epoch:
+            reload_dataloaders_every_n_epochs = reload_dataloaders_every_epoch
+            rank_zero_warn("'reload_dataloaders_every_epoch' is deprecated. Use reload_dataloaders_every_n_epochs", DeprecationWarning)
```

Comment:
```suggestion
            rank_zero_warn(
                "`reload_dataloaders_every_epoch` is deprecated in v1.2 and will be removed in v1.4."
                " Please use `reload_dataloaders_every_n_epochs` in Trainer", DeprecationWarning
            )
```

Reply:
Also add a deprecation here: https://github.com/PyTorchLightning/pytorch-lightning/tree/master/tests/deprecated_api. you need to create a new file for 1.4.

Reply:
Also add the deprecation test here: https://github.com/PyTorchLightning/pytorch-lightning/tree/master/tests/deprecated_api. you need to create a new file for 1.4.